### ISSUE TITLE
Allow All Partial Blocks to Anchor On Barriers

### ISSUE THREAD

**Issue by libbybapa** [BEEmod/BEE2-items#1558]
Just as the title says, it would be really helpful if the partial blocks could anchor on barriers.  When 	"CanAnchorOnBarriers"   "1" then the blocks will actually position on the barrier without causing an error, but the back side is nodraw and the alignment with the barrier is off so they are either suspended away from the barrier or sunk into it ~1/16 voxel.

**Comment by Endermage77** [comment]
It would be really nice to be able to use those 1/4 walls as real walls...
I wonder if there will be a free-standing world-portal door like the regular doors?

**Comment by libbybapa** [comment]
While I appreciate your enthusiasm Endermage, neither of those items are the subject of this issue.  Maybe your concerns would be better voiced in another thread.

**Comment by Endermage77** [comment]
Sorry about that. I must admit, when you said "barriers" I thought that definition would extend to "anything that places like glass," which includes 1/4 walls as far as I'm aware. Whoops.

**Comment by libbybapa** [comment]
Ah, I see what you're saying.  I agree, it would be good to be able to place partial blocks onto the 1/4 wall object.  When I have experimented with other objects anchoring on barriers, ones that have anchored on glass/grating have not anchored on 1/4 walls.

**Comment by Byzarru** [comment]
This manner of functionality should only extend to the quarter wall. In the case of glass and grating, they should offset or angle the barrier accordingly.

**Comment by libbybapa** [comment]
No...  The anchor side of the partial or angled block should anchor to the glass or grating.  The glass or grating should be unmoved/unaltered.

**Comment by Byzarru** [comment]
No... angled glass is a useful aesthetic element and blocks resting on glass looks sloppy. I've previously opened up a thread in which angled and partial blocks effects items, and this would be a logical way to work with glass. My intangible scaffold idea would probably be a better way to position angled and partial blocks (this seems to me to be why you want these items to anchor on barriers).

**Comment by libbybapa** [comment]
Whatever...  That's not what I'm asking for.  I think you should reconsider the idea that my aesthetic opinions are 'sloppy'.  Far from it.  As such I would like the functionality that this thread proposes, *not* what you propose.  Angled glass would be a different item entirely.

**Comment by libbybapa** [comment]
I have seen your thread on having angled blocks affect items.  I hope that is not implemented (I should probably say as much).  No thanks.  It would ruin more than one of my maps.

**Comment by LautaroL20** [comment]
Unfortunately, implementing this feature on Partial Blocks (specifically on Carl Kenner´s ones) will cause a constant resetting on their positions each time you load your puzzle or just when you rotate the barrier itself. See this: https://github.com/BEEmod/BEE2-items/issues/793

By the way, the Tin Wall item is not compatible with the "Anchor" feature since it would keep saying that the attached item (in this case, all Partial Blocks) are still floating. 

Feel free to check all this by yourself if you want.

**Comment by TeamSpen210** [comment]
It'd be applicable to angled or flat partial blocks though.

**Comment by LautaroL20** [comment]
But the problem is that the "Tin Wall" item is not compatible with the "Anchor" feature since it would keep saying that the attached item (in this case, all Partial Blocks) are still floating. So, applying blocks on Glass or Grating woudln´t look that good....

**Comment by Luke18033** [comment]
@LautaroL20 Glass and Grating can be attached to other Glass and Grating, it works fine. Since Thin Wall is placed the same, allowing it to anchor on barriers would also work fine.

**Comment by libbybapa** [comment]
Lautaro, I've seen you make several comments in various threads saying that functionality should not be implemented because 'it would not look good'.  You are certainly far from the best judge of what looks good and attempting to prevent functionality because you lack a quality sense of aesthetics or the creative vision to see how it could be useful is something you should really reconsider.  I have seen several situations where being able to anchor partial blocks to glass or grating would be both useful AND aesthetically pleasing.

**Comment by LautaroL20** [comment]
@Luke18033 - It looks like you and Spencer didn´t play atention to what i wrote, try placing Flat Blocks on Thin Walls then send me a screenshot to see if it works or it just says "Invalid Position".

@libbybapa - Yes, of course since placing concrete blocks on glass makes much sense....

**Comment by libbybapa** [comment]
Lautaro, do you really want/need me to post up some screenshots just to show how incredibly stupid you are being?  It's a waste of my time, but I can do it easily if you need me to.

**Comment by LautaroL20** [comment]
I was talking to @Luke18033, and i didn´t asked you anything, so relax and calm down.

**Comment by libbybapa** [comment]
You're right, you didn't ask me anything, you just replied with '@libbybapa' just before a snide, sarcastic, and stupid comment...

**Comment by LautaroL20** [comment]
Not as so stupid like your ridiculously sentence from about 5 lines adobe one of my comments.

**Comment by libbybapa** [comment]
Your comments are doing a much better job at proving my point than anything I could say.  Thanks.

**Comment by libbybapa** [comment]
Here you go.  Here is a simple situation where it is my firm opinion that finishing the angle of the wall to a point would look better than leaving it truncated like that.  Maybe that is not a situation you would use in one of your maps, but the inability to conceive of situations where others might want to do so is shortsighted indeed.

![20170209091020_1](https://cloud.githubusercontent.com/assets/19520567/22791766/b4416b3e-eea7-11e6-9cc5-f8d443f2ee08.jpg)
![20170209091007_1](https://cloud.githubusercontent.com/assets/19520567/22791778/ba99655e-eea7-11e6-86e7-12dd862c0e0a.jpg)

Is one situation enough for you to alter your closed-minded and unimaginative view?  I can give you more situations if necessary.

**Comment by LautaroL20** [comment]
@Luke18033 - Here is a GIF that proves the Thin Walls can´t anchor on glass or other items, the one that actually can is the original PeTi Glass btw:

![GIF](https://cloud.githubusercontent.com/assets/12244376/22791864/84bad8a6-eec9-11e6-811c-b8d95fe78903.gif)


@libbybapa - Feel free to think whaetever, i do not really care. Ill don´t lose more of my time replying your next issues / comments in the future. I've already pointed out that i don´t think placing concrete blocks on glass or grating would make much sense.

**Comment by libbybapa** [comment]
Thank you.  I would very much appreciate if you did not post or alter the threads that I post.  Your presence in this thread has been entirely anti-helpful.

**Comment by libbybapa** [comment]
What do you think of the screenshots I posted?  You still think no one would ever benefit from being able to anchor partial blocks on barriers?  Lautaro, if you have actual useful information to add, then your input is appreciated.  If all you are is closed-minded and heavy-handed then please do not post in or alter a thread that I start.  Thanks.

**Comment by libbybapa** [comment]
'CanAnchorOnBarriers' works fine for allowing the 1/4 wall to anchor on glass/grating but it does not allow other objects that can anchor on barriers will not anchor on the 1/4 wall.  

![20170209101955_1](https://cloud.githubusercontent.com/assets/19520567/22794767/91e46db6-eeb1-11e6-8d03-11da0d50f48f.jpg)
